[12.0] Add shopinvader_partner_firstname

simahawk opened the pull request:
Integrates `partner_firstname`.

simahawk committed:
[12.0] Add shopinvader_partner_firstname

lmignon reviewed (commented):
At first read, LGTM

simahawk commented:
rebased on #469 as it contains a fix for master and to be merged soon

simahawk commented:
test to fix https://travis-ci.org/shopinvader/odoo-shopinvader/jobs/607689374#L3457

simahawk committed:
Fix shopinvader.partner duplicate test

Remove multiple space between first and lastname
as when this code runs w/ partner_firstname installed
the name will be cleaned up and additional space will be removed
causing the duplication check to fail.

simahawk commented:
@lmignon good for you?

simahawk commented:
hmm I don't get why tests fail only in one of the jobs https://travis-ci.org/shopinvader/odoo-shopinvader/jobs/608155216?utm_medium=notification&utm_source=github_status

simahawk committed:
Partner duplicate: improve `_is_same_partner_value` check

simahawk commented:
@lmignon @sebastienbeau can you give me your opinion regarding this? 
https://github.com/shopinvader/odoo-shopinvader/pull/475/commits/6196b937437b9efd8503a1a45da9c2c50d2748c9
To me it seems more complete and it fixes the case whereas the cache does not contain one of the values.

lmignon commented on shopinvader/models/shopinvader_partner.py:
```diff
@@ -90,14 +90,24 @@ def _is_same_partner_value(self, partner, vals):
                 continue
+            keys_to_check.append(key)
+            if key not in partner._cache:
+                partner[key]  # load missing key
```
:champagne:

simahawk commented:
last build issues fixed, we can merge as it gets green.

simahawk commented:
LINT tests seem to have some warnings for many modules but not for mine. I think we can safely merge and then address warnings if needed.
I'm surprised to see a red light indeed.

lmignon commented on shopinvader/models/shopinvader_partner.py:
```diff
@@ -90,14 +90,24 @@ def _is_same_partner_value(self, partner, vals):
                 continue
+            keys_to_check.append(key)
+            if key not in partner._cache:
+                partner[key]  # load missing key
```
```suggestion
                # pylint: disable=pointless-statement
                partner[key]  # load missing key
```
@simahawk IMO pylint is failing due to this line https://travis-ci.org/shopinvader/odoo-shopinvader/jobs/611453281#L355
![image](https://user-images.githubusercontent.com/544090/68875291-e1551500-0702-11ea-8081-bd6b770570a0.png)

lmignon commented on shopinvader/models/shopinvader_partner.py:
```diff
@@ -90,14 +90,24 @@ def _is_same_partner_value(self, partner, vals):
+            if key in skip_keys or key not in partner:
                 continue
+            keys_to_check.append(key)
+            if key not in partner._cache:
```
@simahawk We don't have to check if the key is already into the cache... it's done by the orm
https://github.com/odoo/odoo/blob/12.0/odoo/fields.py#L989

simahawk commented:
all :green_heart: finally :)

simahawk commented:
/ocabot merge patch

lmignon commented:
:champagne:
